src/arns.lua

```diff
@@ -605,14 +689,13 @@ end
 --- @param initiator string The address of the initiator of the auction
 --- @return Auction|nil The auction instance
 function arns.createAuction(name, timestamp, initiator)
-	assert(arns.getRecord(name), "Name is not registered. Auctions can only be created for registered names.")
+	assert(not arns.getRecord(name), "Name is registered. Auctions can only be created for unregistered names.")
+	assert(not arns.getReservedName(name), "Name is reserved. Auctions can only be created for unregistered names.")
 	assert(not arns.getAuction(name), "Auction already exists for name")
 	local baseFee = demand.baseFeeForNameLength(#name)
 	local demandFactor = demand.getDemandFactor()
 	local auction = Auction:new(name, timestamp, demandFactor, baseFee, initiator, arns.calculateRegistrationFee)
 	NameRegistry.auctions[name] = auction
-	-- ensure the name is removed from the registry
-	arns.removeRecord(name)
 	return auction
```

Comment:
quite the shift of logic here. summing up:

before the name had to be registered, and now we ensure its NOT registered or reserved. and we leave the name on the contract when we create the auction

is this all intended?

Reply:
intended changes - it is now expected that the name will have been removed before an auction can be created from the caller. in this instance, `createAuction` is called from `pruneState` and other handlers.